Update modules

khoahp opened the pull request:
Update module DataItemMgt:
- Update Taglib DataItem
- Update UserMgt:
- Update service.xml
- Declare portlet ID 2 (Reggister User) Portlet ID 3 (Manager User)
- Update AccountMgt:
- Add service.xml
- Declare portlet ID 5 (Register Account), Portlet ID 7 (Manager Account)
- Define package, class, JSP file, language file for portlets.

khoahp committed:
Update service

khoahp committed:
Remove service.properties

khoahp committed:
Update .gitignore

khoahp committed:
Merge remote-tracking branch 'upstream/develop' into develop

khoahp committed:
Add service file for module UserMgt

khoahp committed:
Add JSP and declare User Management Portlet

trungnt-ksnb committed:
Merge pull request #25 from khoahp/develop

Init module UserManagement

trungnt-ksnb committed:
Change package org.opencps.usermgt.portlet.UserManagementPortlet to
org.opencps.usermgt.portlet
Rename clss UserMgtAdminPortlet to UserMgtPortlet

trungnt-ksnb committed:
Remove sample portlet

trungnt-ksnb committed:
Remove sample portlet controler

trungnt-ksnb committed:
Rename jsp folder user_management to usermgt, remove view.jsp file, add
employee.jsp file

trungnt-ksnb committed:
Add toolbar, toptabs, and workingunits file

trungnt-ksnb committed:
Update portlet config

trungnt-ksnb committed:
add UserMgtUtil class

trungnt-ksnb committed:
Add usermgt permission

trungnt-ksnb committed:
Add usermgt language

trungnt-ksnb committed:
Update model hints

khoahp committed:
Update module User Management

Update service.xml:

* Change name table mapping n-n (workingunit_jobspos ->
opencps_workinguni_jobpos, employee_jobpos -> opencps_employee_jobpos)

Update portlet.xml, liferay-portlet.xml, liferay-display.xml:

* Add Account Infomation Management Portlet in portlet.xml
* Add Account Infomation Management Portlet in liferay-portlet.xml
* Edit Portlet Category in liferay-display.xml

Add User Management Portlet:

* Add portlet class
* Add portlet jsp
* Add language file

khoahp committed:
Update service.xml UserMgt

*Add column magagerWorkingUnitId for WorkingUnit entry.

khoahp committed:
Update taglib for DataItem

khoahp committed:
Update init

khoahp committed:
Update service.xml

*Change type database treeIndex from int to String

khoahp committed:
Update service.xml (UserMgt)

* Update  <reference entity="User" package-path="com.liferay.portal" />
for Employee Emtry

khoahp committed:
Update portlet.xml

* Define Accout Register in portlet.xml
* Rename Account Management -> User Management

khoahp committed:
Update module Account Managent

* Backend

- Define service.xml in org.opencps.accountmgt/dao/service.xml
- Define finder, exception in sercive.xml

* Add portlet ID 5 (Account Register)

- Declare portlet in portlet.xml, liferay-portlet.xml
- Group portlet in category, define in liferay-display-portlet.xml
- Define packet and class of portlet
- Define jsp folder of portlet in /accountmgt/account_reg/
- Defile language file in Language_5.properties

* Add portlet ID 7 (Account Management)

- Declare portlet in portlet.xml, liferay-portlet.xml
- Group portlet in category, define in liferay-display-portlet.xml
- Define packet and class of portlet
- Define jsp folder of portlet in /accountmgt/account_reg/
- Defile language file in Language_5.properties

khoahp committed:
Update Module User Management

* Update feature email send tracking:
- Add entry in service.xml

khoahp committed:
Update module Accout Management

* Add entry for Sub Bussines Account

khoahp committed:
Add WorkingUnitFinder

khoahp committed:
Update laguage files

khoahp committed:
Untrack unneeded files

khoahp committed:
Modified Utils Class

khoahp committed:
Add class in org.opencps.usermgt.util.comparator

khoahp committed:
Update util class of UserMgt

khoahp committed:
Module UserMgt

*Add CustomSQL

khoahp committed:
Module UserMgt

*Update class ini ServieImpl

khoahp committed:
Module UserMgt

*Add class Search

khoahp committed:
Module UserMgt

*Update Portlet Class

khoahp committed:
Module UserMgt

*Update Permission

khoahp committed:
Module UserMgt


*Update service.xml

khoahp committed:
Module DataMgt

*Modified util class file

khoahp committed:
Module DataMgt

* Update Data Management Portlet file

khoahp committed:
Module Administrative Procedures

*Add Portlet Class

khoahp committed:
Module Administrative Proceduces

*Add ServieImpl files

khoahp committed:
Module Account Management

*Add Portlet Class for Portlet Edit Account

khoahp committed:
Module AccountMgt

*Update service.xml

khoahp committed:
Update portlet.properties file

khoahp committed:
Update CSS file for portlet

khoahp committed:
Module AccountMgt

*Add JSP files for Module AccountMgt

khoahp committed:
Module ProcedureMgt

*Add JSP for Module ProcedureMgt

khoahp committed:
Module DataMgt

*Update JSP for Module DataMgt

khoahp committed:
Module UserMgt

*Update JSP for portlet User Edit Info

khoahp committed:
Module UserMgt

*Add ajax file

khoahp committed:
Module UserMgt

*Add jsp for Employee Portlet

khoahp committed:
Module UserMgt

*Update jsp for JobPos Portlet

khoahp committed:
Module UserMgt

* Add jsp working unit

khoahp committed:
Module UserMgt

*Update jsp file in UserMgt

khoahp committed:
Module UserMgt

*Update init file

khoahp committed:
Update portlet.xml, liferay-portlet.xml, liferay-display.xml

khoahp committed:
Update default.xml file

khoahp committed:
Add JobPermission file

khoahp committed:
Update hook

khoahp committed:
Update hookc

khoahp commented:
Gửi anh @tanhtanhvn và anh @tuanta nhờ các anh review code để đưa vào nhánh develop.

Các commit cho nhánh này bao gồm:
- Merger module DataMgt
- Merger module UserMgt
- Khai báo các portlet module AccountMgt
- Khai báo các portlet cho module ProcedureMgt

tanhtanhvn commented:
Làm gì có module proceduremgt hả Khoa ơi. Anh đổi tên nó là servicemgt rồi cơ mà. Tên portlet sao không đặt cho nó giống tên module. Khoa quy hoạch lại cái tên portlet để anh xem nào. Hiện đã có 12 cái portlet rồi đấy.
Câu hỏi nữa là merge từ kho cá nhân của Khoa thế này thì sau này có merge được với các nhánh branch nữa không. Mỗi module tạo thành nhánh riêng để sau này merge vào develop. Việc chuẩn hóa code trên các nhánh này hiện nay như thế nào rồi? Có lẽ TuanTA phải ngồi với Khoa 1 vài ngày để chuẩn hóa vấn đề này chăng?

khoahp commented:
@tanhtanhvn 
_Làm gì có module proceduremgt hả Khoa ơi?_
Module này em đặt tên sai mất rồi, em sẽ chỉnh lại và cập nhật lên đây.

_Tên portlet sao không đặt cho nó giống tên module?_
Trong một module có nhiều portlet, hiện tại portlet Id mình đặt theo số, tên module sử dụng cho việc đặt tên các packet của java và folder jsp của portlet.

_Khoa quy hoạch lại cái tên portlet để anh xem nào. Hiện đã có 12 cái portlet rồi đấy._

Em sẽ gửi lại cho anh danh sách portlet, packet khai báo portlet, packet chứa các ServiceImpl, packet chứa service.xml, và folder chứa jsp của từng portlet.

_Câu hỏi nữa là merge từ kho cá nhân của Khoa thế này thì sau này có merge được với các nhánh branch nữa không?_

Tại kho cá nhân của em, em thực hiện việc sync với các nhánh đã hoàn thiện trên kho cá nhân của các bạn.  

Code này được merge từ nhiều nhánh khác nhau. Sau đó em đẩy lên kho Git cá nhân của mình và Tạo pull request đến nhánh develop của kho trung tâm.

Việc merger code từ các nhánh khác nhau theo em cần qua một máy làm nhiệm vụ, merge và pull request vào nhánh develop. Hiện tại em đang làm việc này khi merge các nhánh khác nhau vào và pull resuest lên kho trung tâm.

Các bạn khi develop xong các module cũ thực hiện sync lại với kho trung tâm nhánh develop để cập nhật phần mới nhất của project. 

Nhờ anh Tuấn đánh giá về quy trình này ạ.

tuanta commented:
> Tại kho cá nhân của em, em thực hiện việc sync với các nhánh đã hoàn thiện trên kho cá nhân của các bạn.
> Code này được merge từ nhiều nhánh khác nhau. Sau đó em đẩy lên kho Git cá nhân của mình và Tạo pull request đến nhánh develop của kho trung tâm.
> Việc merger code từ các nhánh khác nhau theo em cần qua một máy làm nhiệm vụ, merge và pull request vào nhánh develop. Hiện tại em đang làm việc này khi merge các nhánh khác nhau vào và pull resuest lên kho trung tâm.
> Các bạn khi develop xong các module cũ thực hiện sync lại với kho trung tâm nhánh develop để cập nhật phần mới nhất của project.

Cách này cũng được, nếu anh em cảm thấy tiện hơn.
Thậm chí cách này còn được khuyến khích sử dụng cho 1 team/company đóng góp vào OpenCPS (thay vì từng cá nhân individual)

khoahp commented:
Dạ vâng ạ. PR này còn một số phân cần chỉnh sửa define packet name cho portlet thủ tục hành chính. Sáng mai em sẽ cập nhật commit cho phần này.

tanhtanhvn commented:
Như vậy hiện nay sinh ra các nhánh feature trên github có vẻ là chưa cần thiết. Vậy tạm thời cứ để nó đấy, để khi nào cộng đồng dev vào đông đảo ta sẽ phân chia quản lý tốt hơn để tránh bị dẫm chân nhau khi làm

tuanta commented:
> Như vậy hiện nay sinh ra các nhánh feature trên github có vẻ là chưa cần thiết. Vậy tạm thời cứ để nó đấy, để khi nào cộng đồng dev vào đông đảo ta sẽ phân chia quản lý tốt hơn để tránh bị dẫm chân nhau khi làm

Cũng không hẳn @tanhtanhvn ạ!

Các nhánh features vẫn cần thiết và recommend các đội fork kho riêng làm theo cho thống nhất.
Sau khi feature DONE họ sẽ merge vào nhánh develop trên kho riêng của nhóm và pull request vào nhánh develop trên kho trung tâm. Như vậy chủ động hơn cho nhóm!
Trường hợp nhiều nhóm cùng dev feature đó trên nhiều kho fork riêng (VD: cả FDS và Bisco) thì lại cần có nhánh feature riêng trên kho trung tâm để 2 nhóm pull request vào đó trước, sau dó mới merge lên nhánh develop => tóm lại là rất flexible, nhưng vẫn bám nguyên tắc ít can thiệp (gần như zero) thẳng vào nhánh develop.

khoahp committed:
Module ServiceMgt

*Add Portlet Class

khoahp committed:
Module UserMgt

*Add portlet class UserContact

khoahp committed:
Module ServiceMgt

*Add JSP file

khoahp committed:
Update porlet.xml and liferay-display.xml files

khoahp committed:
Update language files

khoahp committed:
Update language files

khoahp committed:
Remove files AdminProcedurePortlet and ProceduraPortlet

khoahp committed:
Update portlet.xml file

*Change language location of PortletId 4

khoahp committed:
Modulde ServiceMgt

*Remove folder admtproce
*Add itit.jsp file

khoahp committed:
Remove unneeded file /docroot/view.jsp

*This file automatic generate bay puligin SDK

khoahp commented:
@tuanta, @tanhtanhvn @trungnt-ksnb 

Hiện tại về cấu trúc khung cho dự án đã được chỉnh lý lại, tài liệu về cấu trúc code này em đã gửi cho các anh qua email vào sáng nay, vậy các anh review lại một lần nữa các thay đổi để cho code vào nhánh develop ạ.

tanhtanhvn commented:
@khoahp 
Anh đã cập nhật lại bản kế hoạch và naming các portlet. Khoa rà soát lại lần cuối theo bản kế hoạch này. Chú ý là đối với portlet số 1, hiện nay giao diện đang làm chưa đúng với thiết kế. Do vậy em đổi nó thành portlet số 99 để dùng dự phòng. Như vậy đối vơi module datamgt sẽ có 2 portlet là số 1 và 99. Số 99 sẽ bí xóa đi khi có portlet số 1. Phần taglib thì giữ nguyên để sử dụng.

khoahp committed:
Module AccountMgt

*Add JSP Files init for portlet 5,6,8
*Remove unused files in folder html/accountmgt

khoahp committed:
Module ServiceMgt

*Add JSP file for portlet: 9,10,11
*Remove unused files

khoahp committed:
Add JSP file for portlet 9

khoahp committed:
Add jsp file for portlet 4

khoahp committed:
Add jsp file for Portlet ID 1

khoahp committed:
Update XML file define portlet

khoahp committed:
Module ServiceMgt

* Add service file
* Add *LocalSeviceImpl

khoahp committed:
Module ServiceMgt

* Add JSP files for ddmin service

khoahp committed:
Module ServiceMgt

* Add init file for portlet admin

khoahp committed:
Module ServiceMgt

* Add ServiceUtil for portlet

khoahp committed:
Module ServiceMgt

* Add permission class

khoahp committed:
Module UserMgt

* Add Custom SQL

khoahp committed:
Update resource-action file, add action key for jobpos

khoahp committed:
Module UserMgt

* Update service file, add finder

khoahp committed:
Module UserMgt

*Add resource action class

khoahp committed:
Update language files

khoahp committed:
Module UserMgt

*Update portlet class

khoahp committed:
Module UserMgt

* Add *ServiceImpl

khoahp committed:
Module UserMgt

* Update class utils

khoahp committed:
Module UserMgt

* Update resource action class

khoahp committed:
Module UserMgt

*Update util class in package

khoahp committed:
update portlet.properties, add new key

khoahp committed:
Update JSP files

khoahp committed:
Update service.xml of AccountMgt

khoahp committed:
Module ServiceMgt

*Add JSP files

khoahp committed:
Module ServiceMgt

*Add Permission Files

trungnt-ksnb committed:
Syn last version usermgt

khoahp committed:
Merge pull request #1 from trungnt-ksnb/khoahp-develop

Syn last version usermgt

khoahp committed:
Modulde UserMgt

* Update JSP Files

trungnt-ksnb committed:
Add hook languge

trungnt-ksnb committed:
Merge commit 'FETCH_HEAD' into feature-accountmgt

khoahp committed:
Update UserMgt

*Update action

khoahp committed:
Update WebKeys

*Add key for ServiceMgt

trungnt-ksnb committed:
Merge commit 'FETCH_HEAD' into feature-accountmgt

khoahp committed:
Update ServiceMgt

* Update JSP Files

trungnt-ksnb committed:
Merge commit 'FETCH_HEAD' into feature-accountmgt

khoahp committed:
Update file init.jsp for ServiceMgt

trungnt-ksnb committed:
Merge commit 'FETCH_HEAD' into feature-accountmgt

khoahp committed:
Update jsp folder ServiceMgt

trungnt-ksnb committed:
Merge commit 'FETCH_HEAD' into feature-accountmgt

khoahp committed:
Add ServiceDisplayTerms

khoahp committed:
Update ServiceUtil.java

khoahp committed:
Add service in category ServiceMgt

trungnt-ksnb committed:
Merge commit 'FETCH_HEAD' into feature-accountmgt

trungnt-ksnb committed:
Add taglib multiple upload

trungnt-ksnb committed:
Add taglib handle class multiple upload

trungnt-ksnb committed:
Add class handle DLFolder

trungnt-ksnb committed:
Add TDL file. Definition multiple taglib

khoahp committed:
Update service.xml, update reference to AssetEntry

khoahp committed:
Add Directory Class Portlet

khoahp committed:
Change xml file in META-INF

khoahp committed:
Update CSS file

khoahp committed:
Update JSP file of ServiceMgt module

khoahp committed:
Update liferay-portlet.xml and portlet.xmk

trungnt-ksnb committed:
Add constants account status

khoahp committed:
Update idNumber in service.xml file (AccountMgt)

khoahp committed:
Add hooks

khoahp committed:
Merge pull request #3 from trungnt-ksnb/feature-accountmgt

PR. Taglib multiple upload and class handle DLFolder

tuanta commented:
Pull request này lâu thế mà vẫn chưa merge được vào à?

khoahp commented:
@tanhtanhvn anh merge vào ạ.
